lib/matplotlib/axes/_axes.py

```diff
@@ -4673,8 +4677,9 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
         if not self._hold:
             self.cla()
 
-        if norm is not None:
-            assert(isinstance(norm, mcolors.Normalize))
+        if norm is not None and not isinstance(norm, mcolors.Normalize):
```

Comment:
This one goes against the spirit of duck typing.  Probably fine to leave thought.